Examples/Scripts/Python/material_validation_core.py

```diff
@@ -90,33 +94,132 @@ def runMaterialValidation(s, ntracks, surfaces, outputFile, seed, loglevel):
         help="Construct experimental geometry",
     )
 
-    args = p.parse_args()
+    p.add_argument(
+        "--geomodel-input",
+        type=str,
+        default="",
+        help="Construct experimental geometry from GeoModel",
+    )
 
-    decorators = None
-    if args.map != "":
-        decorators = acts.IMaterialDecorator.fromFile(args.map)
+    p.add_argument(
+        "--geomodel-top-node",
+        type=str,
+        default="",
+        help="Top node definition of the GeoModel tree",
+    )
 
-    if args.experimental:
-        odd_xml = getOpenDataDetectorDirectory() / "xml" / "OpenDataDetector.xml"
+    p.add_argument(
+        "--geomodel-table-name",
+        type=str,
+        default="ActsBlueprint",
+        help="Name of the blueprint table",
+    )
 
-        # Create the dd4hep geometry service and detector
-        dd4hepConfig = DD4hepGeometryService.Config()
-        dd4hepConfig.logLevel = acts.logging.INFO
-        dd4hepConfig.xmlFileNames = [str(odd_xml)]
-        dd4hepGeometryService = DD4hepGeometryService(dd4hepConfig)
-        dd4hepDetector = DD4hepDetector(dd4hepGeometryService)
+    p.add_argument(
+        "--geomodel-queries",
+        nargs="+",
+        type=str,
+        default=[],
+        help="Queries for published GeoModel nodes",
+    )
 
-        cOptions = DD4hepDetectorOptions(logLevel=acts.logging.INFO, emulateToGraph="")
-        cOptions.materialDecorator = decorators
+    args = p.parse_args()
+    gContext = GeometryContext()
+    logLevel = acts.logging.INFO
 
-        # Context and options
-        geoContext = acts.GeometryContext()
-        [detector, contextors, store] = dd4hepDetector.finalize(geoContext, cOptions)
+    materialDecorator = None
+    if args.map != "":
+        materialDecorator = acts.IMaterialDecorator.fromFile(args.map)
 
-        materialSurfaces = detector.extractMaterialSurfaces()
+    if args.experimental:
+        if len(args.geomodel_input) > 0:
+            # Read the geometry model from the database
+            gmTree = acts.geomodel.readFromDb(args.geomodel_input)
```

Comment:
There seems to be quite a lot of duplicated code here in `material_mapping_core.py` and `material_validation_core.py`. We could consider factoring this in a function / functions (e.g., under `Examples/Python/python/acts/examples`)

Reply:
Yes, eventually I will provide a `loadGeomtryGeoModel` function or so, but this will come in another PR.